generator/transit_generator.hpp

```diff
@@ -15,16 +23,35 @@ namespace routing
 {
 namespace transit
 {
+using OsmIdToFeatureIdsMap = std::map<osm::Id, std::vector<FeatureId>>;
+
 class DeserializerFromJson
 {
 public:
-  DeserializerFromJson(json_struct_t * node) : m_node(node) {}
+  DeserializerFromJson(json_struct_t* node, std::shared_ptr<OsmIdToFeatureIdsMap> const & osmIdToFeatureIds);
 
   template<typename T>
   typename std::enable_if<std::is_integral<T>::value || std::is_enum<T>::value || std::is_same<T, double>::value>::type
       operator()(T & t, char const * name = nullptr)
   {
-    GetField(t, name);
+    if (name == nullptr || strcmp(name, "osm_id") != 0)
+    {
+      GetField(t, name);
```

Comment:
I strongly suggest you to use OsmId struct and implement deserialization for OsmId individually. This strcmp looks ugly.

Reply:
Yes. I'll be better I guess. I'll implement that. Thanks.